Notices

simplyNastya opened the pull request.

simplyNastya committed:
Created future routes

refactored code

simplyNastya committed:
Make Notice schema and model

Refactored code

simplyNastya committed:
Update models/notice.js

Refactored code

simplyNastya committed:
Created all controllers for notices

Refactored code

simplyNastya committed:
Refactored code

simplyNastya committed:
Update notices-controllers.js

Fixed getting all user`s notices

simplyNastya committed:
ReFACTORED CODE

simplyNastya committed:
Refactored code

simplyNastya committed:
Update find by category and title

simplyNastya committed:
Refactored code

simplyNastya committed:
Refactored favorite

simplyNastya committed:
Refactored code

simplyNastya committed:
fixed routes

simplyNastya committed:
Added controller to get list of all notices

DaddyStannis commented:
Все супер, тільки декілька правок )) 

1. Видали поле favorite з моделі 
`favorite: {
        type: Boolean,
        default: false
    },`
2. Додавай поле favorite тільки в response json в залежності від того чи є notice id  в массиві user'а. Треба буде перебрати масив notices. Наприклад так:
`const data = notices.map((notice) => {
  const favorite = user.favorites.includes(notice._id);
  return { ...notice, favorite };
});`
3. Можеш перемістити константу `// для додавання оголошень відповідно до обраної категорії
const petAvatarsDirPath = path.resolve("public", "petPhotos");
` вверху файлу, під імпортами?) 
4. Стосовно цієї перевірки при додаванні, та видаленні favorite `const user = await User.findById(_id);
    if (!user) {
      return res.status(404).json({ message: 'The user is not found' });
    }`  в тебе ж юзер і так є в `req.user`. Тут треба перевірити перед пушом чи дійсно існує notice з таким id. Бо помилково можна запушити якесь непотрібне значення.

simplyNastya committed:
Refactored code

Fixed Notice model; replace little bit upper const petAvatarsDirPath; cancel check up for user in the functions add/removeToFavorites and add checking for noticeId

simplyNastya committed:
Update notices-controllers.js

controller findNotices filtered notices by category, title, keyword, sex and birth now

simplyNastya committed:
Update notices-controllers.js

Refactored code

simplyNastya committed:
Refactored code

Added quality to image

simplyNastya committed:
Update Notice model

simplyNastya committed:
Refactored code

simplyNastya committed:
Refactored code

DaddyStannis committed:
Merge branch 'main' into notices
